lib/udf_findhotel_utils.rb

```diff
@@ -292,6 +292,35 @@ def get_bid_match_type(items):
             example: true
           }
         ]
+      },
+      {
+          type:        :function,
+          name:        :make_gha_click_batch_id,
+          description: "Returns a unique identifier for a batch of clicks reported by Google Hotel Ads.",
+          params:      "click_type varchar(max), date_type varchar(max), google_site varchar(max), country varchar(max), device varchar(max)",
+          return_type: "varchar(max)",
+          body:        %~
+            import hashlib
+            import json
+
+            key = {
+                "click_type": click_type,
+                "date_type": date_type,
+                "google_site": google_site,
+                "country": country,
+                "device": device}
+
+            m = hashlib.md5()
+            m.update(json.dumps(key, sort_keys=True).encode())
+            return m.hexdigest()
+
+          ~,
+          tests:       [
+                           {query: "select ?('Standard', 'default', 'localuniversal', 'BR', 'mobile')", expect: '38114881b75e5f9ac8ee9f6b61253084' , example: true},
+                           {query: "select ?('Standard', 'default', 'localuniversal', 'BR', 'mobile')", expect: '38114881b75e5f9ac8ee9f6b61253084' , example: true},
```

Comment:
The above two tests are identical.

Reply:
Yep, I usually add this just to make sure that the function is deterministic, always returning the same hash for the same input.